packages/integrations/vue/src/index.ts

```diff
@@ -39,71 +33,55 @@ function getJsxRenderer(): AstroRenderer {
 	};
 }
 
-function virtualAppEntrypoint(options: ViteOptions) {
+function virtualAppEntrypoint(options?: Options): Plugin {
 	const virtualModuleId = 'virtual:@astrojs/vue/app';
 	const resolvedVirtualModuleId = '\0' + virtualModuleId;
-	let getExports: (id: string) => Promise<string[]>;
+
+	let isBuild: boolean;
+	let root: string;
+
 	return {
 		name: '@astrojs/vue/virtual-app',
-		buildStart() {
-			if (!getExports) {
-				getExports = async (id: string) => {
-					const info = await this.load.call(this, { id });
-					return info.exports ?? [];
-				};
-			}
+		config(_, { command }) {
+			isBuild = command === 'build';
 		},
-		configureServer(server) {
-			if (!getExports) {
-				getExports = async (id: string) => {
-					const mod = await server.ssrLoadModule(id);
-					return Object.keys(mod) ?? [];
-				};
-			}
+		configResolved(config) {
+			root = config.root;
 		},
 		resolveId(id: string) {
 			if (id == virtualModuleId) {
 				return resolvedVirtualModuleId;
 			}
 		},
-		async load(id: string) {
-			const noop = `export const setup = (app) => app;`;
+		load(id: string) {
 			if (id === resolvedVirtualModuleId) {
-				if (options.appEntrypoint) {
-					try {
-						let resolved;
-						if (options.appEntrypoint.startsWith('.')) {
-							resolved = await this.resolve(
-								fileURLToPath(new URL(options.appEntrypoint, options.root))
-							);
-						} else {
-							resolved = await this.resolve(options.appEntrypoint, fileURLToPath(options.root));
-						}
-						if (!resolved) {
-							// This error is handled below, the message isn't shown to the user
-							throw new Error('Unable to resolve appEntrypoint');
-						}
-						const exports = await getExports(resolved.id);
-						if (!exports.includes('default')) {
-							options.logger.warn(
-								`appEntrypoint \`${options.appEntrypoint}\` does not export a default function. Check out https://docs.astro.build/en/guides/integrations-guide/vue/#appentrypoint.`
-							);
-							return noop;
-						}
-						return `export { default as setup } from "${resolved.id}";`;
-					} catch {
-						options.logger.warn(
-							`Unable to resolve appEntrypoint \`${options.appEntrypoint}\`. Does the file exist?`
-						);
-					}
+				if (options?.appEntrypoint) {
+					const appEntrypoint = options.appEntrypoint.startsWith('.')
+						? path.resolve(root, options.appEntrypoint)
+						: options.appEntrypoint;
+
+					return `\
+import * as mod from ${JSON.stringify(appEntrypoint)};
+						
+export const setup = (app) => {
+	if ('default' in mod) {
+		mod.default(app);
+	} else {
+		${
+			!isBuild
+				? `console.warn("[@astrojs/vue] appEntrypoint \`" + ${JSON.stringify(appEntrypoint)} + "\` does not export a default function. Check out https://docs.astro.build/en/guides/integrations-guide/vue/#appentrypoint.");`
```

Comment:
There's something weird happening with the quotes here. Pretty sure we can simplify this to the following and avoid needing to escape the double quotes from `JSON.stringify`

```suggestion
				? `console.warn('[@astrojs/vue] appEntrypoint ${JSON.stringify(appEntrypoint)} does not export a default function. Check out https://docs.astro.build/en/guides/integrations-guide/vue/#appentrypoint.')`
```

Reply:
Yeah the initial code was being a bit more conservative. I figured `appEntrypoint` could contain `'` so using single quotes may break here, e.g. `/Users/Astro's great/project/src/_app.ts`. Maybe backticks work instead? I initially avoided it because I wanted to preserve the backticks when logging the `appEntrypoint`